File: packages/javascript/lib/node/file/createFileWithDummy.ts
import { open } from 'fs/promises';
import { createFile } from './createFile.js';

export async function createFileWithDummy(filePath: string, size: number): Promise<boolean> {
	if (size === undefined || size === null) {
		throw new Error('Size is required');
	}
	if (size < 0) {
		throw new Error('Size must be 0 or greater');
	}

	try {
		if (size === 0) {
			await createFile(filePath);
			return true;
		}

		const data = await open(filePath, 'w');

		await data.write(Buffer.alloc(1), 0, 1, size - 1);
		await data.close();

		return true;
	} catch {
		return false;
	}
}

File: packages/python/qsu/file/createFileWithDummy.py
from .createFile import createFile


def createFileWithDummy(filePath: str, size: int) -> bool:
	if not size or size < 0:
		raise Exception('Size is required')

	try:
		if size == 0:
			createFile(filePath)
			return True

		with open(filePath, 'wb') as data:
			# Write a single null byte at offset (size - 1), producing a sparse
			# file of `size` bytes (mirrors Node's write at position size - 1).
			data.seek(size - 1)
			data.write(b'\x00')

		return True
	except Exception:
		return False
